19 nn vs cnn explaination

Max-Gamill opened the pull request:
Closes #19

Adds a slide in slides-03 after the dense layer recap and before the convolutional layer explanation to explain why the flattened input is not ideal for images - based on the lack of generalising spatial features that are inherent to CNN's.

1. This is only one slide as I want to avoid adding too much due to time (but feel free to let me know if more is wanted).
2. Please let me know if I should drop mentioning CNN layers entirely (as they are only introduced in the next slide).
3. Also let me know if an image would be preferred (I currently can't think of much other than a FC layer of a e.g. 5x5 grid showing that a pixels local area weights are also shared with others due to being FC).

Commit:
Added perfomance vs dense NN slide

Commit:
specifically mentioned flattening the input

Commit:
changed input to images

EdwinB12 commented:
- Addressing how images are different to tabular data
- Address how flattening is a bad thing to do with images
- Therefore why dense layers aren't ideal. Introduce Convolutions.

Commit:
Added slide images

Commit:
+ 2 slides & content: img data diffs +  img dense

Max-Gamill commented:
Removed previous slide and added 2 more:
- One describing the differences in data from images compared to tables (with a picture on flattening the image data to look similar to tabular data) to lead into dense NN's
- The second showing how dense layers don't focus on local pixel areas, with a lead into conv layers (with a picture illustrating this)

EdwinB12 commented:
Oooft, great slides. Think they're really good additions. 

My only comment is I don't think slide 19 - HIERARCHICAL FEATURE REPRESENTAION fits where it is currently placed? 

@MaxGamill-Sheffield - Do you think a better place for this would be the final slide?
